### PR TITLE
BUG: GH12558 where nulls contributed to normalized value_counts

### LINKED ISSUE #12558: BUG?: value_counts(normalize=True) normalizes over all observations including NaN.
By default, `value_counts` ignores missing values. To get them displayed, one must add the option `dropna=False`. But when one normalizes, they are entering into the denominator even when `dropna=True`. 

#### Code Sample, a copy-pastable example if possible

    s = pd.Series([1,2,3,np.nan, np.nan, np.nan])
    s.value_counts(normalize=True)

    0.1667
    0.1667
    0.1667

#### Expected Output

    0.333
    0.333
    0.333

#### output of ``pd.show_versions()``

pd.show_versions()

INSTALLED VERSIONS
------------------
commit: None
python: 3.4.4.final.0
python-bits: 64
OS: Darwin
OS-release: 15.3.0
machine: x86_64
processor: i386
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8

pandas: 0.17.1
nose: 1.3.7
pip: 8.1.0
setuptools: 20.2.2
Cython: 0.23.4
numpy: 1.10.4
scipy: 0.16.1
statsmodels: None
IPython: 4.0.1
sphinx: 1.3.1
patsy: 0.4.0
dateutil: 2.4.2
pytz: 2015.7
blosc: None
bottleneck: 1.0.0
tables: 3.2.2
numexpr: 2.4.4
matplotlib: 1.4.3
openpyxl: 2.2.6
xlrd: 0.9.4
xlwt: 1.0.0
xlsxwriter: 0.7.7
lxml: 3.4.4
bs4: 4.4.1
html5lib: 0.999
httplib2: None
apiclient: None
sqlalchemy: 1.0.9
pymysql: None
psycopg2: None
Jinja2: 2.8

### PR DESCRIPTION
**Pull request by kawochen**
- [x] closes #12558
- [x] tests added / passed
- [x] passes `git diff upstream/master | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: doc/source/whatsnew/v0.18.1.txt [hunk 1]
```diff
@@ -43,3 +43,5 @@ Performance Improvements
 
 Bug Fixes
 ~~~~~~~~~
+
+- Bug in `Series.value_counts` when `normalize=True` and `dropna=True` where nulls still contributed to the normalized count (:issue:`12558`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.18.1.txt]: use double backticks

**Comment by kawochen** [comment]
@nickeubank tests passed. Does this look OK?

File: pandas/core/groupby.py [hunk 2]
```diff
@@ -2949,8 +2949,18 @@ def value_counts(self, normalize=False, sort=True, ascending=False,
 
         if normalize:
             out = out.astype('float')
-            acc = rep(np.diff(np.r_[idx, len(ids)]))
-            out /= acc[mask] if dropna else acc
+            d = np.diff(np.r_[idx, len(ids)])
+            if dropna:
+                m = ids[lab == -1]
+                if compat.numpy_compat._np_version_under1p8:
```
Comment (jreback) [inline pandas/core/groupby.py]: ``from pandas import _np_version_under1p8`` (do the import at the top)

File: pandas/tests/test_algos.py [hunk 3]
```diff
@@ -435,6 +435,22 @@ def test_value_counts_nat(self):
         tm.assert_series_equal(algos.value_counts(dt), exp_dt)
         # TODO same for (timedelta)
 
+    def test_value_counts_normalized(self):
+        # GH12558
+        s = Series([1, 2, np.nan, np.nan, np.nan])
+        dtypes = (np.float64, np.object, 'M8[ns]')
+        for t in dtypes:
+            s_typed = s.astype(t)
+            result = s_typed.value_counts(normalize=True, dropna=False)
```
Comment (jreback) [inline pandas/tests/test_algos.py]: maybe put right after tests for ``dropna``

**Comment by kawochen** [comment]
done

**Comment by nickeubank** [comment]
Looks great!

**Comment by jreback** [comment]
thanks @kawochen @nickeubank

**Pull request closed without merging**